plugins/hls-code-range-plugin/src/Ide/Plugin/CodeRange.hs

```diff
@@ -42,40 +44,67 @@ import           Ide.Types                            (PluginDescriptor (pluginH
                                                        defaultPluginDescriptor,
                                                        mkPluginHandler)
 import           Language.LSP.Server                  (LspM)
-import           Language.LSP.Types                   (List (List),
+import           Language.LSP.Types                   (FoldingRange (..),
+                                                       FoldingRangeParams (..),
+                                                       List (List),
                                                        NormalizedFilePath,
                                                        Position (..),
                                                        Range (_start),
                                                        ResponseError,
-                                                       SMethod (STextDocumentSelectionRange),
+                                                       SMethod (STextDocumentFoldingRange, STextDocumentSelectionRange),
                                                        SelectionRange (..),
                                                        SelectionRangeParams (..),
                                                        TextDocumentIdentifier (TextDocumentIdentifier),
                                                        Uri)
 import           Prelude                              hiding (log, span)
 
 descriptor :: Recorder (WithPriority Log) -> PluginId -> PluginDescriptor IdeState
-descriptor recorder plId = (defaultPluginDescriptor plId)
-    { pluginHandlers = mkPluginHandler STextDocumentSelectionRange selectionRangeHandler
-    -- TODO @sloorush add folding range
-    -- <> mkPluginHandler STextDocumentFoldingRange foldingRangeHandler
-    , pluginRules = codeRangeRule (cmapWithPrio LogRules recorder)
+descriptor recorder plId =
+  (defaultPluginDescriptor plId)
+    { pluginHandlers =
+        mkPluginHandler STextDocumentSelectionRange selectionRangeHandler
+          <> mkPluginHandler STextDocumentFoldingRange foldingRangeHandler,
+      pluginRules = codeRangeRule (cmapWithPrio LogRules recorder)
     }
 
 data Log = LogRules Rules.Log
 
 instance Pretty Log where
-    pretty log = case log of
-        LogRules codeRangeLog -> pretty codeRangeLog
+  pretty log = case log of
```

Comment:
Please revert all indentation changes. Basically, all code is modified, making code review harder.

Reply:
I have removed all the formatting, will format it after the code is reviewed